repl.lua

```diff
@@ -655,17 +633,36 @@ local bindings = {
 	{ 'meta+v',      function() paste(true) end             },
 	{ 'ctrl+w',      del_word                               },
 }
--- Add bindings for all the printable US-ASCII characters from ' ' to '~'
--- inclusive. Note, this is a pretty hacky way to do text input. mpv's input
--- system was designed for single-key key bindings rather than text input, so
--- things like dead-keys and non-ASCII input won't work. This is probably okay
--- though, since all mpv's commands and properties can be represented in ASCII.
-for b = (' '):byte(), ('~'):byte() do
-	local c = string.char(b)
-	local binding = binding_name_map[c] or c
-	bindings[#bindings + 1] = {binding, function() handle_char_input(c) end}
-end
-add_repl_bindings(bindings)
+
+local function text_input(info)
```

Comment:
Is there a reason why this is a ``local function``? It seems to work fine with just ``function``.

Reply:
Just habit. In theory, "local" means the function is put into a register instead of an entry of the global table, so it's faster and possibly reduces conflicts in the global namespace. Probably doesn't matter here.

Reply:
Oh, that's neat. If it's idiomatic Lua, I might start using that for the other binding functions.